core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -320,7 +321,12 @@ private[spark] class Executor(
 
     val taskId = taskDescription.taskId
     val threadName = s"Executor task launch worker for task $taskId"
-    private val taskName = taskDescription.name
+    val taskName = taskDescription.name
+    val mdcProperties = taskDescription.properties.asScala
+      .filter(_._1.startsWith("mdc.")).map { item =>
+        val key = item._1.substring(4)
+        (key, item._2)
+      }.toMap
```

Comment:
we don't really need to look up the resulting map. Seems calling `toSeq` and keeping it as `Seq[(String, String)]` is good enough.

Reply:
Done